core/src/main/java/org/eclipse/hono/util/TenantDataVolume.java

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.util;
+
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSetter;
+
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Optional;
+
+/**
+ * Data volume definition of the tenant resource limits.
+ */
+public class TenantDataVolume {
+
+    private OffsetDateTime effectiveSince;
+
+    @JsonProperty(RegistryManagementConstants.FIELD_MAX_BYTES)
+    private long maxBytes = -1;
+
+    @JsonProperty(RegistryManagementConstants.FIELD_PERIOD)
```

Comment:
this should be used from `TenantConstants` FMPOV as we define the type in the scope of the `TenantObject` ...